rfd/0194-k8s-rbac-simplification.md

```diff
@@ -0,0 +1,220 @@
+---
+authors: Guillaume Charmes (guillaume.charmes@goteleport.com)
+state: draft
+---
+
+# RFD 0194 - Kubernetes RBAC Simplification
+
+## Required Approvers
+
+- Engineering: @rosstimothy && @hugoShaka && @tigrato
+
+## What
+
+The purpose of this RFD is to propose a simplification of the Kubernetes
+RBAC (Role-Based Access Control) within the Teleport project.
+
+The goal is to streamline the user experience and reduce the complexity of
+getting up and running on day 1 and managing permissions later on.
+
+## Why
+
+Today, the Kubernetes RBAC in Teleport is complex and can be difficult for
+users to manage.
+A proper setup requires some external Kubernetes config and matching role
+setup.
+This complexity leads to misconfigurations and security issues.
+
+Simplifying the RBAC model will help improve usability and security.
+
+### References
+
+A common issue currently is the un-intuitive result of complex rule sets and
+more critically, difficult to get setup on day 1.
+
+- @klizhentas struggling to setup a cluster following the `self-hosted` flow:
+  (internal) <https://gravitational.slack.com/archives/C03SULRAAG3/p1715394587181739>
+- Customer confusion on role resolution:
+  - <https://github.com/gravitational/teleport/issues/45475>
+  - (internal) <https://goteleport.slack.com/archives/C0582MBSMHN/p1738187497666819>
+
+## Goals
+
+- Make the Teleport agent authoritative
+  - Handle permissions for the users instead of relying on impersonating a
+    different user/group
+- Discourage (or even deprecate?) the use of the `kubernetes_groups` and
+  `kubernetes_users` in favor of `kubernetes_resources`
+- Clarify / Improve documentation for the various ways to setup Kubernetes in
+  Teleport.
+
+## Proposal
+
+### Background
+
+Currently, when enrolling a Kubernetes cluster and setting it up to use the
+role `system:masters`, everything works as expected. Teleport is super admin
+and yields reduced permissions to the user based on the `kubernetes_resources`
+and `kubernetes_label` fields from the Teleport role.
+
+### Impersonation
+
+To limit the number of code changes, the proposal is to default the
+`kubernetes_group` value to `system:masters` and hide it from the role.
+This can be done by allowing the field to be `null`, and treating `null` as
```

Comment:
I am afraid this will create some issues. Some people assign `access` role - `kubernetes_labels: *:*` but without specifying the `kubernetes_group` or `kubernetes_group`. 
This results in user not having access to the cluster while following your suggestion he would become `cluster-admin`.

Reply:
That will create issues for users when setting up access for Kubernetes because it's not easy to follow what's going on